highway/crud_channel.go

```diff
@@ -3,20 +3,53 @@ package highway
 import (
 	context "context"
 
+	ctv1 "github.com/sonr-io/blockchain/x/channel/types"
+	ot "github.com/sonr-io/blockchain/x/object/types"
 	"github.com/sonr-io/core/channel"
 	ct "go.buf.build/grpc/go/sonr-io/sonr/channel"
 )
 
 // CreateChannel creates a new channel.
 func (s *HighwayServer) CreateChannel(ctx context.Context, req *ct.MsgCreateChannel) (*ct.MsgCreateChannelResponse, error) {
+	// Create ctv1 message to broadcast
+	var fields map[string]*ot.ObjectField
+
+	fields = make(map[string]*ot.ObjectField, len(req.GetObjectToRegister().GetFields()))
```

Comment:
we might want to do some upfront checks before creation of this map wrapper and log some errors / short circuit. otherwise we might be panic on channel creation which could cause further problems.

Reply:
Added check to make sure fields are not nil